src/compas/datastructures/mesh/mesh.py

```diff
@@ -486,7 +491,36 @@ def to_vertices_and_faces(self):
         """
         key_index = self.key_index()
         vertices = [self.vertex_coordinates(key) for key in self.vertices()]
-        faces = [[key_index[key] for key in self.face_vertices(fkey)] for fkey in self.faces()]
+
+        if not triangulated:
+            faces = [[key_index[key] for key in self.face_vertices(fkey)] for fkey in self.faces()]
+            return vertices, faces
+
+        faces = []
+
+        for fkey in self.faces():
+            face_vertices = self.face_vertices(fkey)
+
+            if len(face_vertices) == 3:
+                a, b, c = face_vertices
+                faces.append([key_index[a], key_index[b], key_index[c]])
+            elif len(face_vertices) == 4:
+                a, b, c, d = face_vertices
+                faces.append([key_index[a], key_index[b], key_index[c]])
+                faces.append([key_index[a], key_index[c], key_index[d]])
```

Comment:
her we could potentially check for the "better" option.
but perhaps that is for another PR :)